gateway-service-impl/src/main/java/org/hypertrace/gateway/service/entity/EntityService.java

```diff
@@ -289,6 +304,43 @@ public BulkUpdateEntitiesResponse bulkUpdateEntities(
     return edsEntityUpdater.bulkUpdateEntities(request, updateExecutionContext);
   }
 
+  public BulkUpdateAllMatchingEntitiesResponse bulkUpdateAllMatchingEntities(
+      final RequestContext requestContext, final BulkUpdateAllMatchingEntitiesRequest request) {
+    final BulkUpdateAllMatchingEntitiesUpdateRequestValidator validator =
+        new BulkUpdateAllMatchingEntitiesUpdateRequestValidator(
+            metadataProvider, request, requestContext);
+
+    final Status status = validator.validate();
+    if (!status.isOk()) {
+      LOG.error("Bulk update entities request is not valid: {}", status.getDescription());
+      throw status.asRuntimeException();
+    }
+
+    final Injector injector = Guice.createInjector(new EntityConverterModule());
+    final Converter<BulkUpdateAllMatchingEntitiesRequest, BulkUpdateAllMatchingFilterRequest>
+        requestConverter =
+            injector.getInstance(
+                Key.get(
+                    new TypeLiteral<
+                        Converter<
+                            BulkUpdateAllMatchingEntitiesRequest,
+                            BulkUpdateAllMatchingFilterRequest>>() {}));
+    final Converter<BulkUpdateAllMatchingFilterResponse, BulkUpdateAllMatchingEntitiesResponse>
+        responseConverter =
+            injector.getInstance(
+                Key.get(
+                    new TypeLiteral<
+                        Converter<
+                            BulkUpdateAllMatchingFilterResponse,
+                            BulkUpdateAllMatchingEntitiesResponse>>() {}));
+
+    final BulkUpdateAllMatchingFilterResponse response =
```

Comment:
Should we have a try/catch here?

Reply:
It is there in the caller of this method.